gulp-tasks/test.js

```diff
@@ -0,0 +1,40 @@
+const gulp = require('gulp');
+const mocha = require('gulp-spawn-mocha');
+const path = require('path');
+
+const constants = require('./utils/constants');
+const packageRunnner = require('./utils/package-runner');
+const logHelper = require('./utils/log-helper');
+const pkgPathToName = require('./utils/pkg-path-to-name');
+
+const runBundledTests = (packagePath, env) => {
+  logHelper.log(`Running Test Bundles for ` +
+    `${logHelper.highlight(pkgPathToName(packagePath))}.`);
+
+  const bundleTestDirectory = path.join(
+    packagePath, 'test', constants.BUNDLE_BUILD_DIRNAME);
+  return gulp.src(path.join(bundleTestDirectory, env, '*.js'), {read: false})
+  .pipe(mocha());
+};
+
+gulp.task('test:node', gulp.series(
+  packageRunnner(runBundledTests, 'node'),
```

Comment:
The above is equivalent to:
```javascript
gulp.task('test:node', gulp.series(
  [
    () => runBundledTests('/some-path/workbox/packages/workbox-core', 'node'),
    () => runBundledTests('/some-path/workbox/packages/workbox-precaching', 'node'),
    () => runBundledTests('/some-path/workbox/packages/workbox-*', 'node'),
  ]
)
```

I believe `gulp.series` and `gulp.parallel` will then manage running the functions and chaining Promises / gulp tasks accordingly.

When defining the gulp.task like so:

```javascript
gulp.task('<task name>', <input>);
```

The input variable needs to be a function returning a promise or gulp stream, and since `packageRunner` returns an array, it won't work. `gulp.series` and `gulp.parallel` however will return one of these for us.

I did consider if we should have `packageRunner.series()` and `packageRunner.parallel()` just to hide away the gulp calls, but wasn't sure if that was a good idea or not